Source/VolunteerReporting/TextMessaging/TextMessageProcessor.cs

```diff
@@ -48,131 +47,83 @@ public class TextMessageProcessor : ICanProcessTextMessage
-                    femalesAges0To4,
-                    femalesAgedOver4,
-                    message.Sent);
-            }
-            else
-            {
-                var malesAges0To4 = parsingResult.Numbers[1];
-                var malesAgedOver4 = parsingResult.Numbers[2];
-                var femalesAges0To4 = parsingResult.Numbers[3];
-                var femalesAgedOver4 = parsingResult.Numbers[4];
-                Report(
-                    message,
-                    dataCollector,
-                    caseReporting,
-                    healthRiskId,
-                    malesAges0To4,
-                    malesAgedOver4,
-                    femalesAges0To4,
-                    femalesAgedOver4,
-                    message.Sent);
-            }
-        }
-
-        void ReportInvalidMessage(
-            TextMessage message,
-            IEnumerable<string> errorMessages,
-            DataCollector dataCollector,
-            CaseReporting caseReporting,
-            DateTimeOffset timestamp)
-        {
-            if (dataCollector != null)
                 caseReporting.ReportInvalidReport(
-                dataCollector.Id,
-                message.OriginNumber,
-                message.Message,
-                errorMessages,
-                timestamp);
-            else caseReporting.ReportInvalidReportFromUnknownDataCollector(
-                message.OriginNumber,
-                message.Message,
-                errorMessages,
-                timestamp);
-        }
-
-        void Report(
-            TextMessage message,
-            DataCollector dataCollector,
-            CaseReporting caseReporting,
-            Guid healthRiskId,
-            int malesUnder5,
-            int malesOver5,
-            int femalesUnder5,
-            int femalesOver5,
-            DateTimeOffset timestamp)
-        {
-            if (dataCollector != null)
-            {
-                caseReporting.Report(
                     dataCollector.Id,
-                    healthRiskId,
                     message.OriginNumber,
-                    malesUnder5,
-                    malesOver5,
-                    femalesUnder5,
-                    femalesOver5,
-                    dataCollector.Location.Longitude,
-                    dataCollector.Location.Latitude,
-                    timestamp
-                );
+                    message.Message,
+                    parsingResult.ErrorMessages,
+                    message.Sent);                
+                return;
             }
-            else
+
+            var healthRiskReadableId = parsingResult.HealthRiskReadableId;
+            var healthRiskId = _healthRisks.GetIdFromReadableId(healthRiskReadableId);
+            if (healthRiskId == HealthRiskId.NotSet)
+            {
+                var errorMessages = new List<string> { $"Unable to find health risk, since there are no health risks with a readable id of {healthRiskReadableId}" };
```

Comment:
errorMessages is not used, shouldent this list be appended to parsingResult.ErrorMessages and passed as argument to caseReporting.Report* ?

Reply:
Ups :) Nice catch! Copy paste error